sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

```diff
@@ -145,29 +151,65 @@ private Mono<HttpResponse> attemptAsync(final HttpPipelineCallContext context, f
-                        LOGGER.atInfo()
-                            .addKeyValue(LoggingKeys.TRY_COUNT_KEY, tryCount)
-                            .log("Retry attempts have been exhausted.");
+                        logRetryExhausted(tryCount);
                     }
                     return Mono.just(httpResponse);
                 }
             })
             .onErrorResume(Exception.class, err -> {
                 if (shouldRetryException(err, tryCount)) {
-                    LOGGER.atVerbose()
-                        .addKeyValue(LoggingKeys.TRY_COUNT_KEY, tryCount)
-                            .log("Error resume.", err);
+                    logRetryWithError(LOGGER.atVerbose(), tryCount, "Error resume.", err);
                     return attemptAsync(context, next, originalHttpRequest, tryCount + 1)
                         .delaySubscription(retryStrategy.calculateRetryDelay(tryCount));
                 } else {
-                    LOGGER.atError()
-                        .addKeyValue(LoggingKeys.TRY_COUNT_KEY, tryCount)
-                        .log("Retry attempts have been exhausted.", err);
+                    logRetryWithError(LOGGER.atError(), tryCount, "Retry attempts have been exhausted.", err);
                     return Mono.error(err);
                 }
             });
     }
 
+    private HttpResponse attemptSync(final HttpPipelineCallContext context, final HttpPipelineNextSyncPolicy next,
+                                     final HttpRequest originalHttpRequest, final int tryCount) {
+        context.setHttpRequest(originalHttpRequest.copy());
+        context.setData(HttpLoggingPolicy.RETRY_COUNT_CONTEXT, tryCount + 1);
+        HttpResponse httpResponse;
+        try {
+            httpResponse = next.clone().processSync();
+        } catch (RuntimeException err) {
+            Throwable throwable = Exceptions.unwrap(err);
+            if (shouldRetryException(throwable, tryCount)) {
+                logRetryWithError(LOGGER.atVerbose(), tryCount, "Error resume.", throwable);
+                try {
+                    Thread.sleep(retryStrategy.calculateRetryDelay(tryCount).toMillis());
+                } catch (InterruptedException ie) {
+                    throw LOGGER.logExceptionAsError(new RuntimeException(ie));
+                }
+                return attemptSync(context, next, originalHttpRequest, tryCount + 1);
+            } else {
+                logRetryWithError(LOGGER.atError(), tryCount, "Retry attempts have been exhausted.", throwable);
+                throw LOGGER.logExceptionAsError(err);
+            }
+        }
+        if (shouldRetry(httpResponse, tryCount)) {
+            final Duration delayDuration = determineDelayDuration(httpResponse, tryCount, retryStrategy,
+                retryAfterHeader, retryAfterTimeUnit);
+            logRetry(tryCount, delayDuration);
+
+            httpResponse.getBodyAsBinaryData();
+            httpResponse.close();
```

Comment:
We just need to close it.

```suggestion
            httpResponse.close();
```